README.md

```diff
@@ -765,6 +766,105 @@ NB: Using the JSON Messsage Generator option it is possible to format messages d
+
+#### Add a rule set
+Click `rule sets` on the bottom left and create a new rule set with whatever name you like. If you want to have your function triggered by emails sent to any address in your domain you can leave recipient blank, else you can add the recipient you want to trigger the function.
+
+In that rule set add an action for your Lambda function. This tells AWS to run your action when an email is received by SES.
+
+This is all you need to do to make your function trigger from SES, but depending on what your Lambda function does, it might be kind of hard to verify that it's run (though you can look at the stats for the function to see how and when it's been run) so let's make it save the body of your email to your bucket!
+
+#### Save the email to S3
+Go back to the rule set you created for your Lambda function. Add a new action to it, this one should be an S3 action, with the bucket you want to use selected. This will save the email to S3. Make sure this action is positioned **above** your Lambda function:
+
+![ses management console - google chrome_006](https://user-images.githubusercontent.com/22300773/28177094-a0e16bfc-67f1-11e7-8676-feabc437295f.png)
+
+This saves the email in a slightly weird way that we don't have that much control over, so we want our Lambda function to take that go into the file written from the S3 action and process it separately.
+
+So now you need to set up a role to give your function access to S3, for the policy of the role put something like this:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "logs:CreateLogGroup",
+            "Resource": "arn:aws:logs:us-east-2:YOUR-USER-ID-HERE:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogStream",
+                "logs:PutLogEvents",
+                "s3:*" // This allows all S3 functions
+            ],
+            "Resource": [
+                "arn:aws:logs:us-east-2:271861816104:log-group:/aws/lambda/getHelloWorld:*",
+                "arn:aws:s3:::YOUR-BUCKET-HERE", // Gives access to your bucket
+                "arn:aws:s3:::YOUR-BUCKET-HERE/*" // Gives access to all directories within your bucket
+            ]
+        }
+    ]
+}
+```
+
+As detailed in the comments this allows all s3 functionality for any functions with this role, and gives them access to your bucket and any subdirectories in that bucket.
+
+In the permissions tab of your S3 bucket select "Any Authenticated AWS User" and give read and write access for object access and permission access.
+
+(I'm sure there is a better way than this, __please__ raise an issue if you know how :pray:)
+
+So now when we receive an email it is save to S3 and a function is triggered. Next we need we need to hook the two up so our Lambda function reads the saved file, and saves the body in a nice readable way.
+
+#### Save the body to S3 using AWS Lambda
```

Comment:
If we save all incoming emails it's possible for a SPAMer to flood our Lambda and cause us to waste _loadz-a-money_ on S3 writes ... 
![image](https://user-images.githubusercontent.com/194400/28249477-7f486464-6a4e-11e7-983b-cc7735876ef2.png)
**$0.05 per 1,000** writes to S3 might not _sound_ like a lot, but it will add up if you consider the volume of spam sent/received. (_if you use GMail you won't realise it because Google has **amazing** filters, but an "un-protected" email address will get hundreds and a  reasonably popular domain will get **thousands** of SPAM  emails **per day**_)

Reply:
I've added a note with some of the above comment before the S3 section. Should hopefully help to deter people from leaving this set up on their own accounts. 

A better exercise could be to process the email with Lambda _before_ saving to S3 so we could run something more interesting like a check for a certain keyword or something like that.